packages/language-picker/src/language-picker.tsx

```diff
@@ -77,27 +113,96 @@ const LanguagePicker = ( { onSelectLanguage, languages, languageGroups }: Props
 			};
 
 			return (
-				<div key={ languageGroup.id }>
-					<Button onClick={ onClick } className="language-picker-component__language-group">
-						<span className={ isSelected ? 'is-selected' : '' }>{ languageGroup.name() }</span>
-					</Button>
-				</div>
+				<Button
+					key={ languageGroup.id }
+					onClick={ onClick }
+					className="language-picker-component__language-group"
+				>
+					<span className={ isSelected ? 'is-selected' : '' }>{ languageGroup.name() }</span>
+				</Button>
 			);
 		} );
 	};
 
+	const shouldDisplayRegions = ! search;
+
+	const languagesToRender = search
+		? getSearchedLanguages( languages, search, localizedLanguageNames )
+		: getFilteredLanguages();
+
+	const selectControlOptions = languageGroups.map( ( lg ) => ( { key: lg.id, name: lg.name() } ) );
+
 	return (
-		<div className="language-picker-component">
-			<div className="language-picker-component__regions-label">
-				{ __( 'regions', __i18n_text_domain__ ) }
-			</div>
-			<div className="language-picker-component__content">
-				<div className="language-picker-component__category-filters">
-					{ renderCategoryButtons() }
+		<Flex className="language-picker-component" align="left">
+			<FlexBlock className="language-picker-component__heading">
+				<Flex justify="space-between" align="left">
+					<FlexItem className="language-picker-component__title wp-brand-font">
+						{ headingTitle || __( 'Select a language', __i18n_text_domain__ ) }
+					</FlexItem>
+					{ headingButtons && (
+						<FlexItem className="language-picker-component__heading-buttons">
+							{ headingButtons }
+						</FlexItem>
+					) }
+				</Flex>
+			</FlexBlock>
+			<FlexBlock className="language-picker-component__search">
+				<CustomSelectControl
+					label={ __( 'regions' ) }
+					hideLabelFromVision
+					value={ selectControlOptions.find( ( option ) => option.key === filter ) }
+					options={ selectControlOptions }
+					onChange={ ( { selectedItem } ) => selectedItem && setFilter( selectedItem.key ) }
+				/>
+				<div className="language-picker-component__search-mobile">
+					<Search
```

Comment:
There is an odd popping when opening the search on mobile; you can see that the "Select a Language" title moves for a little before returning to its place.  I haven't investigated what it is, but it seems like the search bar gets too wide for a second when opening.

Reply:
@tyxla Do you mind recording your screen with this happening? I can't reproduce it at the moment.

Reply:
Would love to do that, but for some reason I'm getting an error when trying to build with the latest branch:

```
@automattic/data-stores: src/auth/actions.ts(4,24): error TS7016: Could not find a declaration file for module '@wordpress/data-controls'
```

Are you able to reproduce that?

Reply:
It looks like the CI is failing on this as well.

Reply:
This should be fixed now that search is only 240px wide on desktop. Let me know if you can still reproduce it @tyxla, thanks! 🚀

Reply:
Can't reproduce anymore, I think it's working well now. The only nit I have here is that while the search is opening, there's a scroll appearing. It disappears once the search box ends in its place. Is this something we can easily prevent?